src/main/java/io/vertx/core/Context.java

```diff
@@ -244,6 +246,47 @@ default List<String> processArgs() {
    */
   boolean removeLocal(Object key);
 
+  /**
+   * Get some local data from the context.
+   *
+   * @param key  the key of the data
+   * @param <T>  the type of the data
+   * @return the data
+   */
+  @GenIgnore
+  <T> T getLocal(ContextKey<T> key);
+
+  /**
+   * Get some local data from the context.
+   *
+   * @param key  the key of the data
+   * @param <T>  the type of the data
+   * @return the data
+   */
+  @GenIgnore
+  <T> T getLocal(ContextKey<T> key, Supplier<? extends T> supplier);
```

Comment:
Documentation for `supplier` missing :-) Also it would be a good idea to mention that `supplier` may be called multiple times, which is sometimes important to know.

Reply:
yes, it is actually pretty much like `Map#compute(Function)` that says _When multiple threads attempt updates, map operations and the remapping function may be called multiple times._